src/psyclone/generator.py

```diff
@@ -349,6 +349,17 @@ def generate(filename, api="", kernel_paths=None, script_name=None,
         for invoke in psyir.walk(AlgorithmInvokeCall):
             invoke_trans.apply(
                 invoke, options={"kernels": kernels[id(invoke)]})
+        if api == "dynamo0.3":
+            # Remove any use statements that were temporarily added to
+            # avoid the PSyIR complaining about undeclared builtin
+            # names.
+            for node in psyir.walk((Routine, Container)):
+                symbol_table = node.symbol_table
+                try:
+                    symbol = symbol_table.lookup(builtins_module_name)
```

Comment:
You could do `if builtins_module_name in symbol_table:` here and then you don't need the try..except?

Reply:
OK, although we now lookup the symbol twice.